metaflow/plugins/kfp/kfp.py

```diff
@@ -222,7 +271,7 @@ def _create_workflow_yaml(
             # self.name is typically CamelCase as it's python class name.
             # generateName contains a sanitized version of self.name from kfp.compiler
             workflow["metadata"]["name"] = (
-                workflow["metadata"].pop("generateName").rstrip("-")
+                name if name else workflow["metadata"].pop("generateName").rstrip("-")
```

Comment:
nit: Might be less readable but can use `or`: `name or workflow["metadata"].pop("generateName").rstrip("-")`